logstash.go

```diff
@@ -27,21 +29,58 @@ func NewHookWithConn(conn net.Conn, appName string) (*Hook, error) {
 // NewHookWithFields creates a new hook to a Logstash instance, which listens on
 // `protocol`://`address`. alwaysSentFields will be sent with every log entry.
 func NewHookWithFields(protocol, address, appName string, alwaysSentFields logrus.Fields) (*Hook, error) {
+	return NewHookWithFieldsAndPrefix(protocol, address, appName, alwaysSentFields, "")
+}
+
+func NewHookWithFieldsAndPrefix(protocol, address, appName string, alwaysSentFields logrus.Fields, prefix string) (*Hook, error) {
 	conn, err := net.Dial(protocol, address)
 	if err != nil {
 		return nil, err
 	}
-	return NewHookWithFieldsAndConn(conn, appName, alwaysSentFields)
+	return NewHookWithFieldsAndConnAndPrefix(conn, appName, alwaysSentFields, prefix)
 }
 
 // NewHookWithFieldsAndConn creates a new hook to a Logstash instance using the supplied connection
 //The new Hook
 func NewHookWithFieldsAndConn(conn net.Conn, appName string, alwaysSentFields logrus.Fields) (*Hook, error) {
-	return &Hook{conn: conn, appName: appName, alwaysSentFields: alwaysSentFields}, nil
+	return NewHookWithFieldsAndConnAndPrefix(conn, appName, alwaysSentFields, "")
+}
+
+func NewHookWithFieldsAndConnAndPrefix(conn net.Conn, appName string, alwaysSentFields logrus.Fields, prefix string) (*Hook, error) {
+	return &Hook{conn: conn, appName: appName, alwaysSentFields: alwaysSentFields, hookOnlyPrefix: prefix}, nil
+}
+
+func (h *Hook) filterHookOnly(entry *logrus.Entry) {
```

Comment:
I don't like this name (`filterHookOnly`). Lets rename it to `removePrefixFields`.